integration/spark/src/main/scala/org/apache/spark/sql/optimizer/CarbonOptimizer.scala

```diff
@@ -72,23 +74,71 @@ object CarbonOptimizer {
 class ResolveCarbonFunctions(relations: Seq[CarbonDecoderRelation])
   extends Rule[LogicalPlan] with PredicateHelper {
   val LOGGER = LogServiceFactory.getLogService(this.getClass.getName)
-  def apply(plan: LogicalPlan): LogicalPlan = {
-    if (relations.nonEmpty && !isOptimized(plan)) {
+  def apply(logicalPlan: LogicalPlan): LogicalPlan = {
+    if (relations.nonEmpty && !isOptimized(logicalPlan)) {
+      val plan = processPlan(logicalPlan)
+      val udfTransformedPlan = pushDownUDFToJoinLeftRelation(plan)
```

Comment:
Why plan change is required? I don't think it is better idea to do plan changes in spark.